kraken/node_actions/common_node_functions.py

```diff
@@ -32,17 +32,26 @@ def get_node(node_name, label_selector, instance_kill_count):
 
 # Wait till node status becomes Ready
 def wait_for_ready_status(node, timeout):
-    runcommand.invoke("kubectl wait --for=condition=Ready " "node/" + node + " --timeout=" + str(timeout) + "s")
+    runcommand.invoke("kubectl wait --for=condition=Ready node/" + node + " --timeout=" + str(timeout) + "s")
 
 
 # Wait till node status becomes NotReady
 def wait_for_unknown_status(node, timeout):
     for _ in range(timeout):
-        if kubecli.get_node_status(node) == "Unknown":
-            break
+        try:
+            node_status = kubecli.get_node_status(node, timeout)
+            if node_status is None or node_status == "Unknown":
+                break
+        except Exception:
+            logging.error("Encountered error while getting node status, waiting 3 seconds and retrying")
         time.sleep(3)
-    if kubecli.get_node_status(node) != "Unknown":
-        raise Exception("Node condition status isn't Unknown")
+    try:
+        node_status = kubecli.get_node_status(node, timeout)
+        logging.info("node status " + str(node_status))
+        if node_status is not None and node_status != "Unknown":
+            logging.error("Node condition status isn't Unknown after %s seconds" % str(timeout))
```

Comment:
Should it be an Exception instead of just logging it as "isn't unknown" like before.
`raise Exception("Node condition status isn't Unknown")`